File: fix.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src', 'components', 'pages', 'provisioning', 'modals', 'ProvisionServiceModal.tsx');
let content = fs.readFileSync(filePath, 'utf-8');

// Remove TabType 'packet'
content = content.replace("type TabType = 'general' | 'protocol' | 'packet' | 'access';", "type TabType = 'general' | 'protocol' | 'access';");

// Remove the 'packet' tab from the array
content = content.replace(/\{ id: 'packet' as TabType, label: 'Thiết kế cấu trúc gói tin', icon: <LayoutTemplate className="w-4 h-4" \/> \},?\n\s*/g, "");

// Remove states for packet design
content = content.replace(/  \/\/ States for packet design \(Tab 3\)[\s\S]*?  const handleUpdateJoinTable = \(id: number, key: string, value: string\) => \{[\s\S]*?  \};\n\n/g, "");

// Remove the functions handleAddDataField to generateDynamicPreview
content = content.replace(/  const handleAddDataField = \(\) => \{[\s\S]*?  const generateDynamicPreview = \(\) => \{[\s\S]*?    return JSON\.stringify\(fullResponse, null, 2\);\n  \};\n\n/g, "");

// Remove the TAB 3 JSX block
content = content.replace(/            \{\/\* TAB 3: Thiết kế cấu trúc gói tin \*\/\}[\s\S]*?            \{\/\* TAB 4: Phân quyền truy cập \*\/\}/g, "            {/* TAB 3: Phân quyền truy cập */}");

// Fix the steps footer
content = content.replace(/\{\[1,2,3,4\]\.map\(step => \(/g, "{[1,2,3]}.map(step => (");
content = content.replace(/\(activeTab === 'packet' && step === 3\) \|\| \n/g, "");
content = content.replace(/\(activeTab === 'access' && step === 4\)/g, "(activeTab === 'access' && step === 3)");
content = content.replace(/\{activeTab === 'general' \? 'Step 1 of 4' : activeTab === 'protocol' \? 'Step 2 of 4' : activeTab === 'packet' \? 'Step 3 of 4' : 'Step 4 of 4'\}/g, "{activeTab === 'general' ? 'Step 1 of 3' : activeTab === 'protocol' ? 'Step 2 of 3' : 'Step 3 of 3'}");

// Remove mockSchema
content = content.replace(/\/\/ Mock Database Schema for Civil Registry[\s\S]*?const tableNames = Object\.keys\(mockSchema\);\n\n/g, "");

fs.writeFileSync(filePath, content, 'utf-8');
console.log("Replaced successfully!");

File: fix.py
import re

with open('src/components/pages/provisioning/modals/ProvisionServiceModal.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Remove TabType 'packet'
content = content.replace("type TabType = 'general' | 'protocol' | 'packet' | 'access';", "type TabType = 'general' | 'protocol' | 'access';")

# Remove the 'packet' tab from the array
content = re.sub(r"\{ id: 'packet' as TabType, label: 'Thiết kế cấu trúc gói tin', icon: <LayoutTemplate className=\"w-4 h-4\" /> \},?\n\s*", "", content)

# Remove states for packet design
content = re.sub(r"  // States for packet design \(Tab 3\).*?  const handleUpdateJoinTable = \(id: number, key: string, value: string\) => \{.*?  \};\n\n" , "", content, flags=re.DOTALL)

# Remove the functions handleAddDataField to generateDynamicPreview
content = re.sub(r"  const handleAddDataField = \(\) => \{.*?  const generateDynamicPreview = \(\) => \{.*?    return JSON.stringify\(fullResponse, null, 2\);\n  \};\n\n", "", content, flags=re.DOTALL)

# Remove the TAB 3 JSX block
content = re.sub(r"            \{\/\* TAB 3: Thiết kế cấu trúc gói tin \*\/\}.*?            \{\/\* TAB 4: Phân quyền truy cập \*\/\}", "            {/* TAB 3: Phân quyền truy cập */}", content, flags=re.DOTALL)

# Fix the steps footer
content = re.sub(r"\{\[1,2,3,4\]\.map\(step => \(", "{[1,2,3]}.map(step => (", content)
content = content.replace("(activeTab === 'packet' && step === 3) || \n", "")
content = content.replace("(activeTab === 'access' && step === 4)", "(activeTab === 'access' && step === 3)")
content = content.replace("{activeTab === 'general' ? 'Step 1 of 4' : activeTab === 'protocol' ? 'Step 2 of 4' : activeTab === 'packet' ? 'Step 3 of 4' : 'Step 4 of 4'}", "{activeTab === 'general' ? 'Step 1 of 3' : activeTab === 'protocol' ? 'Step 2 of 3' : 'Step 3 of 3'}")

# Remove mockSchema
content = re.sub(r"// Mock Database Schema for Civil Registry.*?const tableNames = Object\.keys\(mockSchema\);\n\n", "", content, flags=re.DOTALL)

with open('src/components/pages/provisioning/modals/ProvisionServiceModal.tsx', 'w', encoding='utf-8') as f:
    f.write(content)
